### ISSUE TITLE
Upgrading from GO to PT Fails

### ISSUE THREAD

**Issue by valerie-rstudio** [rstudio/rstudio#10547]
### System details

    RStudio Edition : Server & Workbench
    RStudio Version : 2022.02.0-431 & 2022.02.0-preview-431.pro1
    OS Version      : ubuntu20
    R Version       : 4.1.0

### Summary
Upgrading from GO to PT results in users unable to connect to the rstudio server.

### Repro Steps
1. Spin up a fuzzbucket environment with the last official GO patch
2. Verify server is up and you are able to login
3. Download the latest preview of PT onto the environment
4. Suspend any active sessions `sudo rstudio-server suspend-all`
5. Put the server in offline mode `sudo rstudio-server offline`
6. Install the PT version `sudo gdebi <packageName>`
7. Restart the server `sudo rstudio-server restart`
8. Bring the server back online `sudo rstudio-server online`
9. **Expected**: The server is back online and you are able to connect and login.
10. **Actual**: The server status shows no errors `sudo rstudio-server status` and there are no log errors, however the login screen does not appear and the user receives a message that "R is taking longer to start than usual".

### Notes
- Sourced upgrade steps from our official documentation here: https://docs.rstudio.com/rsw/upgrade/#upgrade-procedures
- The way we have our R modules specified in fuzzbucket do cause some status errors, but the above repro steps and issue holds true even when not using R modules and/or upgrading open source.
- Upgrading desktop works as expected

**Comment by melissa-barca** [comment]
I saw something similar recently, but thought it was an issue with my local environment rather than with the product. I believe the key here is running `sudo rstudio-server suspend-all`. I saw this command hanging and not properly completing. Note that if you try to log in via a different browser or incognito mode after the upgrade, you're able to login.

**Comment by valerie-rstudio** [comment]
Yes, just re-ran the test and can confirm that logging in from incognito does work.

**Comment by melissa-barca** [comment]
I'm seeing this warning in the logs:
```
rsession-melissa[50059]: WARNING RPC request to '/rpc/get_active_sessions' has missing or mismatched rs-csrf-token cookie or X-RS-CSRF-Token header; LOGGED FROM: bool rstudio::session::{anonymous}::parseAndValidateJsonRpcConnection(rstudio_boost::shared_ptr<rstudio::session::HttpConnection>, rstudio::core::json::JsonRpcRequest*) src/cpp/session/SessionHttpMethods.cpp:133
```
@jmcphers do you think this could be related to the change that renamed the csrf token cookie/header?

**Comment by melissa-barca** [comment]
Refreshing the browser doesn't fix it and neither does opening the url in a new tab. Even in a new tab, it looks the cookie is missing the `rs-` prefix: 

<img width="959" alt="image" src="https://user-images.githubusercontent.com/5323711/153286705-74552408-52e4-4599-ac30-158588d9e45b.png">

**Comment by jmcphers** [comment]
That makes sense. We probably need to either explicitly rename the cookie or respect the old cookie name if it is present and the new one isn't (this change occurred because of cookie name conflicts with DataBricks).

**Comment by jmcphers** [comment]
Verified fixed in 2022.02.0+438.pro1.

- Installed Ghost Orchid 2021.09.2+382.pro1
- Create a session
- Installed Prairie Trillium 2022.02.0+438.pro1 (Upgrade)
- Verified that session is still accessible
- Verified that `csrf-token` and `rs-csrf-token` are both set
- Installed Ghost Orchid 2021.09.2+382.pro1 (Downgrade)
- Verified that session is still accessible
